### PR TITLE
(PUP-8780) Remove sshkeys_core code from Puppet

### PR DESCRIPTION
**Pull request by jhelwig**
While the `sshkeys_core` module is going to be re-packaged into puppet-agent, there is no guarantee that the type will be available if Puppet is run from source. We now provide a warning if a feature of the
user type that depends on one of the types in the `sshkeys_core` module is used while the type from the `sshkeys_core` is not available.

### CODE REVIEW

File: spec/unit/type/user_spec.rb [hunk 1]
```diff
@@ -457,7 +457,7 @@ def check_valid_shell; end
     end
   end
 
-  describe "when purging ssh keys" do
+  describe "when purging ssh keys", :if => Puppet::Type.type(:ssh_authorized_key) do
```
Comment (kris-bosland) [inline spec/unit/type/user_spec.rb]: Should these tests be moved to the module?

File: spec/integration/type/user_spec.rb [hunk 2]
```diff
@@ -15,7 +15,7 @@
   include PuppetSpec::Files
   include PuppetSpec::Compiler
 
-  context "when set to purge ssh keys from a file" do
+  context "when set to purge ssh keys from a file", :if => Puppet::Type.type(:ssh_authorized_key) do
```
Comment (kris-bosland) [inline spec/integration/type/user_spec.rb]: Should these tests be moved to the module?

File: spec/unit/type/user_spec.rb (hunk 1 above)
Comment (jhelwig) [reply spec/unit/type/user_spec.rb]: Yes, they should. That came up in chat, but hasn't been done yet.

File: spec/integration/type/user_spec.rb (hunk 2 above)
Comment (jhelwig) [reply spec/integration/type/user_spec.rb]: Yes, they should. That came up in chat, but hasn't been done yet.

File: lib/puppet/type/user.rb [hunk 3]
```diff
@@ -675,13 +675,23 @@ def delimiter
     end
 
     def generate
-      return [] if self[:purge_ssh_keys].empty?
-      find_unmanaged_keys
+      if !self[:purge_ssh_keys].empty?
+        if Puppet::Type.type(:ssh_authorized_key).nil?
+          warning _("Ssh_authorized_key type is not available. Cannot purge SSH keys.")
```
[the remark this answers is not in the document]
Comment (jhelwig) [reply lib/puppet/type/user.rb]: They shouldn't be, but there's no guarantees that the module will always be packaged with puppet-agent, or that someone knows they need to install the module if they're running from source. Seemed better to warn than to silently do nothing.

**Comment by joshcooper** [comment]
Also need to remove Sshkey and Ssh_authorized_key from:

```
lib/puppet/pops/loader/static_loader.rb:      Ssh_authorized_key
lib/puppet/pops/loader/static_loader.rb:      Sshkey
spec/unit/pops/loaders/static_loader_spec.rb:      Ssh_authorized_key
spec/unit/pops/loaders/static_loader_spec.rb:      Sshkey
```

**Comment by jhelwig** [comment]
@joshcooper Gah! Thanks for the catch. I always forget about those. :(

**Pull request merged**